Parse entire macsima metadata from OME metadata

MSHelm opened the pull request:
Fixes https://github.com/scverse/spatialdata-io/issues/348

The parser now tries to use the filename stored in the tiff metadata, if it exists. This is more robust, as users typically do not modify these entries.
If this metadata doesnt exist (which is the case for older data), it falls back to the actual filename.

Since the lung dataset used for testing is using an old format, I added a newer data set that is publicy available (https://zenodo.org/records/14008816). I documented this in the tests as well. 

Full disclosure: I am working for Miltenyi, but I love your work ;)

LucaMarconato commented:
Hi @MSHelm, thanks for the contribution! I’ve fixed the unrelated failing tests on main, and your PR is now green.

I had a quick look but before full review I kindly ask you to verify and confirm which items from the [contribution guide checklist](https://github.com/scverse/spatialdata-io/blob/main/docs/contributing.md) apply to this PR. Ensuring the contributions guidelines are followed help streamline our review process and improve long-term code maintainability. Much appreciated!

LucaMarconato commented:
Thank you for the update and for going through the checklist!

> Regarding the test data, I will provide some small versions of OMAP10 and OMAP23 via zenodo for the CI. Should then these be used for the unit tests?
> It seems to me that the large original data is currently not fetched in the CI, and no tests on macsima are run due to this.

Yes please, the small datasets (and only the small datasets) should be included in the CI.

MSHelm commented:
Hey @LucaMarconato 
thank you for the review! As you suggested I added the channel names as a tie breaker for sorting. I thought about sorting according to wavelength of the channel, which is how it is displayed in MACSiQ View, but I didnt find it in the metadata directly. So since it doesnt really add much for now we can keep it simple I think.

Regarding the Lung adc data set, can you point me towards which one you are using for this? I looked into our internal demo data sets and didnt find one that would fit the description directly.

In the end I added an additional check for "MICS cycle ID" as you also suggested. But this was done blind, so it would be great if you could check that with the actual data set ^^"

MSHelm commented:
@LucaMarconato I added the clarifications for the format specs
